### DISCUSSION TITLE
Dynamically set maxDate LockPlugin

### DISCUSSION THREAD

**Discussion by irenefelita192** [easepick/easepick#106] [Q&A] [answered]
Hi, I have two datepicker using easepick, first for start date and second for end date. I do not use Range Plugin for this but create 2 picker instance. I have this case, so if start date is selected, the second datepicker's date before start date is disabled and also 3 months after start date is also disabled.I'm using sveltekit. And if I debug on lockplugins, it already set to new maxDate but the view on the datepicker is not updated.
Here is few line of codes I'm trying to do but not working

```

$: if (lockDate) {
		if (picker) {
			const lockPlugin = picker.PluginManager.getInstance('LockPlugin')

			if (lockDate) {
				lockPlugin.options.maxDate = lockDate
			} else {
				lockPlugin.options.maxDate = null
			}
			picker.renderAll()
		}
	}
let picker
onMount(()=>{
     const lockPlugin = maxDate
			? {
					plugins: ['LockPlugin'],

					LockPlugin: {
						maxDate
					}
			  }
			: {}
		const datepicker = document.getElementById(`datepicker-${id}`)
		if (easepick && datepicker) {
			picker = new easepick.create({
				element: datepicker,
				css: [
					'https://cdn.jsdelivr.net/npm/@easepick/bundle@1.1.5/dist/index.css',
					'/style/datepickerStyle.css'
				],
				date: selectedDate.formatted,
				format: 'DD MMM YYYY',
				...lockPlugin
			})
		}
       }
   }
```

Is this possible or I should use RangePlugin?
Thank you

**Comment by verdant-spark** [comment]
Did you ever get to a resolution for this?

**Comment by nathanmedhurstschoolshr** [comment]
have you tried using: picker.renderAll(); at the end of what you are trying to do?

**Reply by verdant-spark** [reply]
I have an alternative solution; works for me.

**Comment by wakirin** [comment, marked as answer]
https://jsfiddle.net/Ldstwgq6/

can be something like that.
answering just for history because @irenefelita192 probably already found solution or uses another library.

**Reply by irenefelita192** [reply]
This can be another alternative. Currently since I use one datepicker as reusable component, I can make this happen by destroying then set lockdate. 
Thank you for this response

**Reply by verdant-spark** [reply]
This was also my solution. Depending on an option picked elsewhere on the page, the calendar options vary (whether Sundays are available, which days are knocked out, etc) - it's easier to set all the options on click of the other component and then destroy any existing picker and re render the new one.
